File: src/playing_with_digits/java/DigPow.java
package playing_with_digits.java;

public class DigPow {
    public static long digPow(int n, int p) {
        String[] s = String.valueOf(n).split("");
        int expression = 0;
        for (String value : s) {
            expression += (int) (Math.pow(Double.parseDouble(value), p));
            p++;
        }
        int result = expression / n;
        if (expression == n * result) {
            return result;
        }
        return -1;
    }
}

File: src/kyu_6/playing_with_digits/js/digPow.js
function digPow(n, p){
    let s = n.toString().split('');
    let sum = 0;
    for (let i = 0; i < s.length; ++i, ++p)
    sum += Math.pow(s[i], p);
    return (sum % n === 0) ? sum / n : -1;
}

module.exports = digPow;